File: src/main/java/com/example/desafio/api/models/customvalidation/MaxYearValidator.java
package com.example.desafio.api.models.customvalidation;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;
import java.util.Calendar;

public class MaxYearValidator implements ConstraintValidator<MaxYear, Integer> {

    private int currentYear;

    @Override
    public void initialize(MaxYear constraintAnnotation) {
        currentYear = Calendar.getInstance().get(Calendar.YEAR);
    }

    @Override
    public boolean isValid(Integer value, ConstraintValidatorContext context) {
        return value <= currentYear;
    }
}

File: src/main/kotlin/com/example/desafio/api/models/customValidation/MaxYearValidator.kt
package com.example.desafio.api.models.customValidation


import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import java.util.*


val currentYear: Int
    get() = Calendar.getInstance()[Calendar.YEAR]

class MaxYearValidator : ConstraintValidator<MaxYear, Int> {
    override fun isValid(value: Int, context: ConstraintValidatorContext?): Boolean {
        return value <= currentYear
    }
}